From 1f1dd9da522c288a4dda381828854162a65cabf1 Mon Sep 17 00:00:00 2001
From: Ahzo <Ahzo@tutanota.com>
Date: Sat, 31 Jul 2021 13:24:39 +0200
Subject: [PATCH] Adapt to python2 removal

Use ipython3 and cython3 instead of ipython and cython.

Also call ipython3 directly instead of from SAGE_LOCAL, as the latter is
set to /usr anyway in the package.
During the build SAGE_LOCAL points to sage/local, but for some reason
ipython3 is not linked into its bin subfolder.
---
 sage/src/bin/sage                 | 6 +++---
 sage/src/sage/interfaces/tests.py | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/sage/src/bin/sage
+++ b/sage/src/bin/sage
@@ -539,12 +539,12 @@
 
 if [ "$1" = '-ipython' -o "$1" = '--ipython' ]; then
     shift
-    exec "$SAGE_LOCAL"/bin/ipython "$@"
+    exec ipython3 "$@"
 fi
 
 if [ "$1" = '-ipython3' -o "$1" = '--ipython3' ]; then
     shift
-    exec "$SAGE_LOCAL"/bin/ipython3 "$@"
+    exec ipython3 "$@"
 fi
 
 if [ "$1" = '-jupyter' -o "$1" = '--jupyter' ]; then
@@ -558,7 +558,7 @@
 
 if [ "$1" = "-cython" -o "$1" = '--cython' -o "$1" = '-pyrex' -o "$1" = "--pyrex" ]; then
     shift
-    exec cython "$@"
+    exec cython3 "$@"
 fi
 
 if [ "$1" = '-gap' -o "$1" = '--gap' ]; then
--- a/sage/src/sage/interfaces/tests.py
+++ b/sage/src/sage/interfaces/tests.py
@@ -37,7 +37,7 @@
     True
     sage: subprocess.call("echo syntax error | gp", **kwds)
     0
-    sage: subprocess.call("echo syntax error | ipython", **kwds) in (0, 1, 120)
+    sage: subprocess.call("echo syntax error | ipython3", **kwds) in (0, 1, 120)
     True
     sage: subprocess.call("echo syntax error | Singular", **kwds)
     0
